aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1984,4 +1985,29 @@ Tensor movedim(const Tensor& self, int64_t src, int64_t dst) {
   return at::movedim(self, IntArrayRef{src}, IntArrayRef{dst});
 }
 
+Tensor trace_cpu(const Tensor& self) {
+  Tensor result = at::empty({}, self.options());
+  AT_DISPATCH_ALL_TYPES(self.scalar_type(), "trace", [&] {
```

Comment:
can we also enable it for complex dtypes? it should be straightforward to do that (similar to `torch.sum`)

Reply:
I took a quick look. Enabling it for complex dtypes isn't as simple as changing the dispatch macro here; we also have to change the CUDA trace implementation to work with complex.

The CUDA trace implementation requires that torch.diag() works with complex (it doesn't currently).

To keep things simple, I'd like to propose just shipping this as-is and leaving the complex enabling in a follow-up PR